src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/GaugeWave.hpp

```diff
@@ -0,0 +1,164 @@
+ * \partial_y \alpha & = & \partial_z \alpha = 0.
+ * \f} The shift \f$\beta^i\f$ and its derivatives vanish.
+ *
+ * The extrinsic curvature's components vanish, except for
+ * \f{eqnarray}{
+ * K_{xx} & = & \frac{\partial_x H}{2 \sqrt{H}}.
+ * \f}
+ *
+ * The following are input file options that can be specified:
+ *  - Amplitude (default: 1.)
+ *  - Wavelength (default: 1.)
+ */
+class GaugeWave : public MarkAsAnalyticSolution {
+ public:
+  static constexpr size_t volume_dim = 3;
+  struct Amplitude {
+    using type = double;
+    static constexpr OptionString help = {"Amplitude of the gauge wave"};
+    static type default_value() noexcept { return 1.; }
+    static type lower_bound() noexcept { return 0.; }
+  };
+  struct Wavelength {
+    using type = double;
+    static constexpr OptionString help = {"Wavelength of the gauge wave"};
+    static type default_value() noexcept { return 1.; }
+    static type lower_bound() noexcept { return 0.; }
+  };
+
+  using options = tmpl::list<Amplitude, Wavelength>;
+  static constexpr OptionString help{"Gauge wave in flat spacetime"};
+
+  GaugeWave(double amplitude, double wavelength,
+            const OptionContext& context = {});
+
+  explicit GaugeWave(CkMigrateMessage* /*unused*/) noexcept {}
+
+  GaugeWave() = default;
+  GaugeWave(const GaugeWave& /*rhs*/) = default;
+  GaugeWave& operator=(const GaugeWave& /*rhs*/) = default;
+  GaugeWave(GaugeWave&& /*rhs*/) noexcept = default;
+  GaugeWave& operator=(GaugeWave&& /*rhs*/) noexcept = default;
+  ~GaugeWave() = default;
+
+  template <typename DataType>
+  using DerivLapse = ::Tags::deriv<gr::Tags::Lapse<DataType>,
+                                   tmpl::size_t<volume_dim>, Frame::Inertial>;
+  template <typename DataType>
+  using DerivShift =
+      ::Tags::deriv<gr::Tags::Shift<volume_dim, Frame::Inertial, DataType>,
+                    tmpl::size_t<volume_dim>, Frame::Inertial>;
+  template <typename DataType>
+  using DerivSpatialMetric = ::Tags::deriv<
+      gr::Tags::SpatialMetric<volume_dim, Frame::Inertial, DataType>,
+      tmpl::size_t<volume_dim>, Frame::Inertial>;
+  template <typename DataType>
+  using tags = tmpl::list<
+      gr::Tags::Lapse<DataType>, ::Tags::dt<gr::Tags::Lapse<DataType>>,
+      DerivLapse<DataType>,
+      gr::Tags::Shift<volume_dim, Frame::Inertial, DataType>,
+      ::Tags::dt<gr::Tags::Shift<volume_dim, Frame::Inertial, DataType>>,
+      DerivShift<DataType>,
+      gr::Tags::SpatialMetric<volume_dim, Frame::Inertial, DataType>,
+      ::Tags::dt<
+          gr::Tags::SpatialMetric<volume_dim, Frame::Inertial, DataType>>,
+      DerivSpatialMetric<DataType>, gr::Tags::SqrtDetSpatialMetric<DataType>,
+      gr::Tags::ExtrinsicCurvature<volume_dim, Frame::Inertial, DataType>,
+      gr::Tags::InverseSpatialMetric<volume_dim, Frame::Inertial, DataType>>;
+
+  template <typename DataType>
+  tuples::tagged_tuple_from_typelist<tags<DataType>> variables(
```

Comment:
Please make this work for any set of tags, like all analytic solutions except Kerr-Schild do (unless there's a reason not to?). See any of the hydro solutions for examples.

What I mean is to add `variables` overloads for every tag that's supported, each with just a single tag in the `TaggedTuple`. Then add another overload that collects them all:

```cpp
  template <typename... Tags>
  tuples::TaggedTuple<Tags...> variables(const tnsr::I<DataVector, 3>& x,
                                         double t,
                                         tmpl::list<Tags...> /*meta*/) const
      noexcept {
    // precompute data here
    return {get<Tags>(variables(x, t, /*precomputed_data,*/ tmpl::list<Tags>{}))...};
  }
```

You probably want to precompute data to share between the individual `variables` functions. You can do so you by precomputing it in the shared `variables` function (see above) and passing it down to the individual functions. Then you can make these individual functions private. `NewtonianEuler/RiemannProblem.hpp` is an example for this patter.